components/raftstore/src/store/peer.rs

```diff
@@ -702,6 +724,253 @@ pub enum UnsafeRecoveryState {
     Destroy(UnsafeRecoveryExecutePlanSyncer),
 }
 
+// #[RaftstoreCommon], copied from Peer
+/// Peer's trait
+pub trait RaftPeer<EK, ER>
```

Comment:
How about reimplementing the logic instead of using trait? And we should think whether the flow is still valid or whether can be improved.

Reply:
I think for the read function, its core logic does not have real difference for multi-rocks and single rocksdb. Do you have an example which part is invalid or can be improved specifically to multi-rocks version.